cpp/daal/include/algorithms/k_nearest_neighbors/kdtree_knn_classification_predict_types.h

```diff
@@ -107,9 +119,70 @@ class DAAL_EXPORT Input : public classifier::prediction::Input
     services::Status check(const daal::algorithms::Parameter * parameter, int method) const DAAL_C11_OVERRIDE;
 };
 
+/**
+ * <a name="DAAL-CLASS-ALGORITHMS__RIDGE_REGRESSION__PREDICTION__RESULT"></a>
+ * \brief Provides interface for the result of KD-tree based kNN model-based prediction
+ */
+class DAAL_EXPORT Result : public classifier::prediction::Result
+{
+public:
+    DECLARE_SERIALIZABLE_CAST(Result)
+    Result();
+
+    /**
+     * Returns the result of KD-tree based kNN model-based prediction
+     * \param[in] id    Identifier of the result
+     * \return          Result that corresponds to the given identifier
+     */
+    data_management::NumericTablePtr get(ResultId id) const;
+
+    /**
+     * Sets the result of KD-tree based kNN model-based prediction
+     * \param[in] id      Identifier of the input object
+     * \param[in] value   %Input object
+     */
+    void set(ResultId id, const data_management::NumericTablePtr & value);
+
+    /**
+     * Allocates memory for storing prediction results of KD-tree based kNN algorithm
+     * \tparam  algorithmFPType     Data type for storing prediction results
+     * \param[in] input     Pointer to the input objects of the classification algorithm
+     * \param[in] parameter Pointer to the parameters of the classification algorithm
+     * \param[in] method    Computation method
+     */
+    template <typename algorithmFPType>
+    DAAL_EXPORT services::Status allocate(const daal::algorithms::Input * input, const daal::algorithms::Parameter * parameter, int method);
+
+    /**
+     * Checks the correctness of prediction results of KD-tree based kNN algorithm
+     * \param[in] input     Pointer to the the input object
+     * \param[in] parameter Pointer to the algorithm parameters
+     * \param[in] method    Computation method
+     */
+    services::Status check(const daal::algorithms::Input * input, const daal::algorithms::Parameter * parameter, int method) const DAAL_C11_OVERRIDE;
+
+protected:
+    using classifier::prediction::Result::check;
+
+    /** \private */
+    services::Status checkImpl(const daal::algorithms::Input * input, const daal::algorithms::Parameter * parameter) const;
+
+    /** \private */
+    template <typename Archive, bool onDeserialize>
+    services::Status serialImpl(Archive * arch)
+    {
+        return classifier::prediction::Result::serialImpl<Archive, onDeserialize>(arch);
+    }
+};
+typedef services::SharedPtr<Result> ResultPtr;
+typedef services::SharedPtr<const Result> ResultConstPtr;
+
 } // namespace interface1
 
 using interface1::Input;
+using interface1::Result;
```

Comment:
may be, interface3?

Reply:
No, interfaceN are not consistent for any alg anyway, because we add new interfaceN for classes actually changed, so there are no reasons to skip some numbers in interfaceN names.